src/components/CardModal/CardModalAttachments/CardModalAttachment.js

```diff
@@ -0,0 +1,164 @@
+
+    const dispatch = useDispatch();
+    
+
+    useEffect(async()=> {
+        let result = [];
+        await Promise.all(card.attachments.map(async(item) => {
+            let file = await localforage.getItem(item.fileId)
+            let url = URL.createObjectURL(file) 
+            let isImage = true
+            let fileExtension = getFileExtension(file.name)
+            let color = arrColors[randomInt(0,arrColors.length)]
+            if (fileExtension!=='png') {
+                isImage=false
+                color = 'rgb(229 231 235)'
+            }
+            console.log(item.fileId, url)
+            // TODO: replace with dominant color - right now it is random
+            result.push({item, file: url, color, isImage, fileExtension})
+        }))
+        setFiles(result);
+    },[card])   
+
+    const handleShowAttachmentDeleteModal = (id) => {
+        dispatch(showAttachmentDeleteMenu({id}));
+    }
+
+    const handleShowAttachmentEditModal = (id) => {
+        dispatch(showAttachmentEditMenu({id}));
+    }
+
+    const handleShowAttachmentMenuModal = (id) => {
+        dispatch(showAttachmentMenu({id}));
+    }
+
+  // TODO: Add formated date to date span
+  return (<>
+    { files.length > 0 && 
+        (<div className="flex-col">
+            <div className="flex items-center gap-2 mt-2 text-lg mb-2 text-gray-800 font-semibold">
+                <AiOutlinePaperClip size={20} />
+                <p>Attachments</p>
+            </div>
+            { files.map( attachment =>
+                <div key={attachment.item.fileId} className=" flex mb-3  hover:bg-black hover:bg-opacity-5 rounded">
+                    <div className='flex items-center'>
+                        <div className='flex w-32 h-24 items-center justify-center rounded overflow-hidden' style = {{backgroundColor: attachment.color }}> 
+                            <a className="max-w-32 max-h-24 rounded" download={attachment.item.name} href = {attachment.file} >
+                                { attachment.isImage ? (
+                                    <img className = "max-w-32 max-h-24" src = {attachment.file} />
+                                ) : (
+                                    <span className="flex text-lg font-bold text-gray-600">{attachment.fileExtension}</span>
+                                )
+                                }
+                            </a>
+                        </div>
+                    </div>
+                    <div className="flex w-full flex-col text-gray-600 pl-3 rounded-ibs text-sm">  
+                        <span className="font-bold break-all pt-1.5">
+                            {attachment.item.name}
+                        </span>
+                        <span className="break-words">
+                            <span dt={ attachment.item.date}>{ getRelativeDate(attachment.item.date)} </span>
+                            <span> - <span className="underline cursor-pointer">Comment</span></span>
```

Comment:
Should comment be here if we dont have comments functionality? Might be worth creating an issue to add comments

Reply:
https://github.com/rizwan171/trello-clone/issues/138